### ISSUE TITLE
Add support for extracting a bit mask from a Vector<T>

### ISSUE THREAD

**Issue by mburbea** [dotnet/runtime#14291]
SSE2/AVX supports extracting the MSB from each 8 byte,4 byte,2 byte or 1 byte segment (the `movemask_*` family). This can be extremely useful while performing algorithms that are data sensitive and use SSE registers like big buffers. 

A user could implement algorithms that work entirely in safe code that could benefit from avoiding the allocation and rigmarole of copying data out of a register to an array when you just want to test if a field is not empty. 

Admittedly, it is a downer as ARM does not have an intrinsic for this operation but there is workarounds that should perform better than copying everything in and out memory to test the bytes.
http://stackoverflow.com/questions/11870910/sse-mm-movemask-epi8-equivalent-method-for-arm-neon

**Comment by tannergooding** [comment]
Going to close this for now. This hasn't gained much traction and likely needs more thought/scenarios before it could be moved forward.

If someone feels strongly about this, please feel free to open a new issue. Our API review process is detailed [here](https://github.com/dotnet/runtime/blob/master/docs/project/api-review-process.md) with a "good example" being listed under step 1. Ideally the new issue would provide many of the sections given, but would at a minimum provide the base rationale and proposed API surface sections.

For this API in particular, the exposure of the System.Runtime.Intrinsic types and the methods for converting to/from `Vector<T>` should unblock users who immediately need this functionality.
